internal/appsec/waf_test.go

```diff
@@ -426,3 +426,65 @@ func TestBlocking(t *testing.T) {
 		})
 	}
 }
+
+// Test that API Security schemas get collected when API security is enabled
+func TestAPISecurity(t *testing.T) {
+	// Start and trace an HTTP server
+	mux := httptrace.NewServeMux()
+	mux.HandleFunc("/apisec", func(w http.ResponseWriter, r *http.Request) {
+		pAppsec.MonitorParsedHTTPBody(r.Context(), "plain body")
+		w.Write([]byte("Hello World!\n"))
+	})
+	srv := httptest.NewServer(mux)
+	defer srv.Close()
+
+	req, err := http.NewRequest("POST", srv.URL+"/apisec?vin=AAAAAAAAAAAAAAAAA", nil)
+	require.NoError(t, err)
+
+	t.Run("enabled", func(t *testing.T) {
+		t.Setenv("DD_EXPERIMENTAL_API_SECURITY_ENABLED", "true")
+		t.Setenv("DD_API_SECURITY_REQUEST_SAMPLE_RATE", "1.0")
+		appsec.Start()
+		defer appsec.Stop()
+		if !appsec.Enabled() {
+			t.Skip("AppSec needs to be enabled for this test")
+		}
+		mt := mocktracer.Start()
+		defer mt.Stop()
+
+		res, err := srv.Client().Do(req)
+		require.NoError(t, err)
+		defer res.Body.Close()
+
+		spans := mt.FinishedSpans()
+		require.Len(t, spans, 1)
+
+		// Make sure the addresses that are present are getting extracted as schemas
+		require.NotNil(t, spans[0].Tag("_dd.appsec.s.req.headers"))
+		require.NotNil(t, spans[0].Tag("_dd.appsec.s.req.query"))
+		require.NotNil(t, spans[0].Tag("_dd.appsec.s.req.body"))
```

Comment:
These strings are the result of `json.Marshal()` so either:
- marshalling errored out in which case the tag doesn't get set (nil case)
- marshilling succeeded in which case strings won't be blank. At worst it will be a "`{}`" or "`null`". In which case it means schema generation is having issues on the WAF side, which is tested in the waf_unit_test.go file

I can add an in depth check that expects a specific schema. I didn't since these checks are performed in waf_unit_test.go, and this simply verifies that the output gets added to tags when the configuration allows it, regardless of whatever kind of schema change could happen in the future

Reply:
I'm okay with this line of reasoning as long as there is _another_ test that verifies that if the tags are set, their values are "as expected", which... I guess there is?

Reply:
Added some tag specific tests here: 9bf98c80ec702d407127